Sources/Fullscreen/FrontpageSearch/FrontpageSearchView.swift

```diff
@@ -0,0 +1,221 @@
+                    return cell
+                }
+            }
+        )
+
+        dataSource.supplementaryViewProvider = { [weak self] collectionView, kind, indexPath in
+            if kind == "UICollectionElementKindSectionHeader" {
+
+                let view = collectionView.dequeue(
+                    FrontpageSearchSectionHeader.self,
+                    for: indexPath,
+                    ofKind: UICollectionView.elementKindSectionHeader
+                )
+                guard let section = self?.sections[safe: indexPath.section] else {
+                    view.configure()
+                    return view
+                }
+                switch section {
+                case .group(let group):
+                    guard group.items.count > 0 else {
+                        view.configure()
+                        return view
+                    }
+                    view.configure(with: group.title)
+                case .viewMoreResults(title: let title):
+                    view.configure()
+                case .locationPermission(title: let title):
+                    view.configure()
+                }
+                return view
+            } else if kind == "UICollectionElementKindSectionFooter" {
+                let view = collectionView.dequeue(
+                    FrontpageSearchSectionFooter.self,
+                    for: indexPath,
+                    ofKind: UICollectionView.elementKindSectionFooter
+                )
+                return view
+            }
+            return UICollectionReusableView()
+        }
+        return dataSource
+    }()
+
+    private var sections = [FrontpageSearchSection]()
+    private typealias Snapshot = NSDiffableDataSourceSnapshot<FrontpageSearchSection, FrontpageSearchGroupItem>
+    private weak var delegate: FrontpageSearchViewDelegate?
+    private var remoteImageViewDataSource: RemoteImageViewDataSource
+
+    // MARK: - Internal properties
+
+    static let horizontalSpacing: CGFloat = .spacingS
+
+    // MARK: - Init
+
+    public init(
+        withAutoLayout: Bool,
+        delegate: FrontpageSearchViewDelegate?,
+        remoteImageViewDataSource: RemoteImageViewDataSource
+    ) {
+        self.delegate = delegate
+        self.remoteImageViewDataSource = remoteImageViewDataSource
+        super.init(frame: .zero)
+        translatesAutoresizingMaskIntoConstraints = !withAutoLayout
+        setup()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - Setup
+
+    private func setup() {
+        backgroundColor = .bgTertiary
+        addSubview(collectionView)
+        collectionView.fillInSuperview(insets: UIEdgeInsets(top: .spacingM, leading: 0, bottom: 0, trailing: 0))
+    }
+
+    // MARK: - Snapshot management
+
+    public func configure(with sections: [FrontpageSearchSection], delegate: FrontpageSearchViewDelegate?) {
```

Comment:
Is there a need to have the `delegate` argument here? It's already present in the constructor and doesn't seem to change

Reply:
Good catch, testing it without confirmed it continued to work without so I've removed it.